src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/TripPatternForDate.java

```diff
@@ -47,27 +55,52 @@
     public TripPatternForDate(
         TripPatternWithRaptorStopIndexes tripPattern,
         List<TripTimes> tripTimes,
+        List<FrequencyEntry> frequencies,
         LocalDate localDate
     ) {
         this.tripPattern = tripPattern;
         this.tripTimes = new ArrayList<>(tripTimes);
+        this.frequencies = frequencies;
         this.localDate = localDate;
 
-        // These depend on the tripTimes array being sorted
-        this.startOfRunningPeriod = DateMapper.asDateTime(
-            localDate,
-            tripTimes.get(0).getDepartureTime(0)
-        );
-        var last = tripTimes.get(tripTimes.size()-1);
-        this.endOfRunningPeriod = DateMapper.asDateTime(
-            localDate, last.getArrivalTime(last.getNumStops() - 1)
-        );
+        // TODO: We expect a pattern only containing trips or frequencies, fix ability to merge
```

Comment:
perhaps we can assert that this is the case and throw a warning on import, if not done already? It might also be good to add this to tests so we can check that a mixture of frequency and non-frequency data behaves as we expect

Reply:
Good catch. I will add a check + issue store entry in the graph build. If we have a request, the pattern can be split in two parts. one for frequency based trips and one for schedule-based.